nomad/structs/node_pool.go

```diff
@@ -133,3 +138,39 @@ func (n *NodePoolSchedulerConfiguration) Validate() error {
 
 	return mErr.ErrorOrNil()
 }
+
+// NodePoolListRequest is used to list node pools.
+type NodePoolListRequest struct {
+	QueryOptions
+}
+
+// NodePoolListResponse is the response to node pools list request.
+type NodePoolListResponse struct {
+	NodePools []*NodePool
+	QueryMeta
+}
+
+// NodePoolSpecificRequest is used to make a request for a specific node pool.
+type NodePoolSpecificRequest struct {
+	Name string
+	QueryOptions
+}
+
+// SingleNodePoolResponse is the response to a specific node pool request.
+type SingleNodePoolResponse struct {
+	NodePool *NodePool
+	QueryMeta
+}
+
+// NodePoolUpsertRequest is used to make a request to insert or update a node
+// pool.
+type NodePoolUpsertRequest struct {
+	NodePools []*NodePool
+	WriteRequest
+}
+
+// NodePoolDeleteRequest is used to make a request to delete a node pool.
+type NodePoolDeleteRequest struct {
+	Names []string
+	WriteRequest
+}
```

Comment:
These requests take a list to help synchronizing node pools with mutiregion federation.